### PR TITLE
Prune literally all kube-apiserver internal client 

### PR DESCRIPTION
**Pull request by yue9944882**
i dont think this will get into v1.13, so for now, it's just experimental.

/sig api-machienery
/kind cleanup

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: test/integration/auth/rbac_test.go [hunk 1]
```diff
@@ -46,8 +47,7 @@ import (
 	"k8s.io/kubernetes/pkg/api/legacyscheme"
 	"k8s.io/kubernetes/pkg/api/testapi"
 	api "k8s.io/kubernetes/pkg/apis/core"
-	rbacapi "k8s.io/kubernetes/pkg/apis/rbac"
```
Comment (yue9944882) [inline test/integration/auth/rbac_test.go]: @deads2k 😺 i guess it should have been done in #63967

**Comment by lavalamp** [comment]
:) :)

**Comment by yue9944882** [comment]
this pull is so large that it keeps conflicting 😥will split it into smaller ones

File: pkg/registry/scheduling/rest/storage_scheduling.go [hunk 2]
```diff
@@ -106,7 +107,12 @@ func AddSystemPriorityClasses() genericapiserver.PostStartHookFunc {
 				_, err := schedClientSet.PriorityClasses().Get(pc.Name, metav1.GetOptions{})
 				if err != nil {
 					if apierrors.IsNotFound(err) {
-						_, err := schedClientSet.PriorityClasses().Create(pc)
+						// TODO: Remove this explicit conversion after scheduling api move to v1
```
Comment (liggitt) [inline pkg/registry/scheduling/rest/storage_scheduling.go]: these are v1 in https://github.com/kubernetes/kubernetes/pull/73557

scheduling.SystemPriorityClasses() should probably return versioned objects (and the method might need to move to a different package to be able to do that)

File: pkg/registry/scheduling/rest/storage_scheduling.go [hunk 3]
```diff
@@ -30,12 +31,12 @@ import (
 	"k8s.io/apiserver/pkg/registry/rest"
 	genericapiserver "k8s.io/apiserver/pkg/server"
 	serverstorage "k8s.io/apiserver/pkg/server/storage"
+	schedulingclient "k8s.io/client-go/kubernetes/typed/scheduling/v1beta1"
```
Comment (liggitt) [inline pkg/registry/scheduling/rest/storage_scheduling.go]: use the v1 client

File: hack/update-codegen.sh [hunk 4]
```diff
@@ -71,7 +71,6 @@ INTERNAL_DIRS_CSV=$(IFS=',';echo "${INTERNAL_DIRS[*]// /,}";IFS=$)
 
 # This can be called with one flag, --verify-only, so it works for both the
 # update- and verify- scripts.
-${clientgen} --input-base="k8s.io/kubernetes/pkg/apis" --input="${INTERNAL_DIRS_CSV}" "$@"
```
Comment (liggitt) [inline hack/update-codegen.sh]: drop INTERNAL_DIRS_CSV?

**Comment by liggitt** [comment]
a couple nits, but since this is a rebase magnet, we can take those in follow-up PRs

/lgtm
/approve

**Comment by yliaog** [comment]
good to see this done.

/lgtm

File: hack/update-codegen.sh (hunk 4 above)
Comment (yue9944882) [reply hack/update-codegen.sh]: fixing here https://github.com/kubernetes/kubernetes/pull/76363

File: pkg/registry/scheduling/rest/storage_scheduling.go (hunk 2 above)
Comment (yue9944882) [reply pkg/registry/scheduling/rest/storage_scheduling.go]: fixing in this thread https://github.com/kubernetes/kubernetes/pull/76362

File: pkg/registry/scheduling/rest/storage_scheduling.go (hunk 3 above)
Comment (yue9944882) [reply pkg/registry/scheduling/rest/storage_scheduling.go]: > fixing in this thread https://github.com/kubernetes/kubernetes/pull/76362

ditto

**Pull request merged**